gr-zeromq/include/gnuradio/zeromq/pub_sink.h

```diff
@@ -57,6 +57,11 @@ namespace gr {
        */
       static sptr make(size_t itemsize, size_t vlen, char *address,
                        int timeout=100, bool pass_tags=false, int hwm=-1);
+
+      /*!
+       * \brief If the endpoint is TCP return the port number; -1 otherwise.
+       */
+      virtual int port() = 0;
```

Comment:
Currently most of the common implementation code wouldn't have been applicable to the message ones basically ... so it didn't make sense to use the same base class for stream vs message.

And making it a 3 level   base_impl <- base_{stream,message}_impl <- {pub,sub,..}_stream_impl    would have been quite useless because base_impl ended up being empty of any code.

So I opted to just do a separate base class for the stream and left the message ones for someone else to refactor.